SumoLogic.Logging.AspNetCore/LoggerProvider.cs

```diff
@@ -93,17 +100,40 @@ public void WriteLine(String message, String categoryName)
                 return;
             }
 
-            String line = string.Concat(
-                message.TrimEnd(Environment.NewLine.ToCharArray()),
-                Environment.NewLine);
+            var builder = new StringBuilder();
+            if (includeCategory)
+            {
+                builder.Append(" [");
+                builder.Append(logLevel.ToString());
+                builder.Append("] ");
+                builder.Append(categoryName);
+            }
+
+            var scopeProvider = ScopeProvider;
+            if (scopeProvider != null)
+            {
+                scopeProvider.ForEachScope((scope, stringBuilder) =>
+                {
+                    stringBuilder.Append(this.scopeSeparator).Append(scope);
+                }, builder);
+
+                builder.Append(this.scopeSeparator);
+            }
+            else if(includeCategory)
+            {
+                builder.Append(this.scopeSeparator);
+            }
+
+
+            builder.AppendLine(message.Trim());
```

Comment:
Why are we changing the _default_ logic from

```
String line = string.Concat(
                message.TrimEnd(Environment.NewLine.ToCharArray()),
                Environment.NewLine);
```

to just
`message.Trim()`?

Reply:
I'm trying to remember my exact reason - it's been a while :)

The changes were based off of the ConsoleLogger implementation of scopes:
https://github.com/aspnet/Extensions/blob/master/src/Logging/Logging.Console/src/ConsoleLogger.cs

Specifically, they replace all new lines in message, not just the beginning and end. I know I encountered a need for this during testing, but I can't remember the case anymore.

What are your concerns in regards to trimming the start?

Reply:
I am unsure on why the initial code was written like that. There might have been a reason too! :) So I am just worried about touching something which we are not completely aware of. And if we can't justify this change right now, I'll prefer to revert this change. We can obviously add this change later on in some other PR, if needed.